File: SWEA/D2/1959. 두 개의 숫자열/두 개의 숫자열.java
import java.util.Scanner;

class Solution {
    public static void main(String args[]) throws Exception {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();  // 테스트 케이스의 수

        for (int test_case = 1; test_case <= T; test_case++) {
            int N = sc.nextInt();  // Ai의 크기
            int M = sc.nextInt();  // Bj의 크기
            
            int[] Ai = new int[N];
            int[] Bj = new int[M];
            
            // Ai 배열 입력 받기
            for (int i = 0; i < N; i++) {
                Ai[i] = sc.nextInt();
            }

            // Bj 배열 입력 받기
            for (int i = 0; i < M; i++) {
                Bj[i] = sc.nextInt();
            }

            // 더 긴 배열을 기준으로 겹치는 구간을 찾아 곱셈 합산
            int maxSum = Integer.MIN_VALUE;

            // 배열 Bj를 Ai 배열에 맞춰 이동시킴
            for (int offset = 0; offset <= Math.abs(N - M); offset++) {
                int sum = 0;
                // Ai의 i번째와 Bj의 i + offset 번째를 곱해서 더하기
                for (int i = 0; i < Math.min(N, M); i++) {
                    int aiIndex = i + (N > M ? offset : 0);
                    int bjIndex = i + (M > N ? offset : 0);
                    if (aiIndex < N && bjIndex < M) {
                        sum += Ai[aiIndex] * Bj[bjIndex];
                    }
                }
                maxSum = Math.max(maxSum, sum);  // 최대값 갱신
            }

            // 결과 출력
            System.out.println("#" + test_case + " " + maxSum);
        }
    }
}

File: SWEA/D2/1959. 두 개의 숫자열/두 개의 숫자열.py
T = int(input())  # 테스트 케이스 수

# 여러 개의 테스트 케이스가 주어지므로, 각각을 처리합니다.
for test_case in range(1, T + 1):
    N, M = map(int, input().split())  # N과 M 입력 받기
    Ai = list(map(int, input().split()))  # Ai 배열 입력 받기
    Bj = list(map(int, input().split()))  # Bj 배열 입력 받기

    # 더 긴 배열을 기준으로 배치하면서 곱의 합을 계산
    max_sum = -float('inf')  # 최댓값 초기화 (가능한 가장 작은 값으로 시작)

    # 더 긴 배열을 기준으로 이동하면서 계산
    if N >= M:
        for offset in range(N - M + 1):  # Ai 배열을 기준으로 Bj 배열을 이동
            current_sum = 0
            for i in range(M):
                current_sum += Ai[offset + i] * Bj[i]
            max_sum = max(max_sum, current_sum)
    else:
        for offset in range(M - N + 1):  # Bj 배열을 기준으로 Ai 배열을 이동
            current_sum = 0
            for i in range(N):
                current_sum += Ai[i] * Bj[offset + i]
            max_sum = max(max_sum, current_sum)

    print(f"#{test_case} {max_sum}")  # 결과 출력
